src/go/k8s/controllers/redpanda/cluster_controller_test.go

```diff
@@ -79,6 +77,19 @@ var _ = Describe("RedPandaCluster controller", func() {
 			}
 			Expect(k8sClient.Create(context.Background(), redpandaCluster)).Should(Succeed())
 
+			By("Expecting to delete successfully")
+			Eventually(func() error {
```

Comment:
I don't think this needs to be in eventually. Just issue delete and then wait for that to eventually succeed (and please verify the error is notafound)

Reply:
I removed this commit all together.